src/paymasters/SponsorPaymaster.sol

```diff
@@ -23,18 +23,17 @@ import "../interfaces/IKintoWallet.sol";
 contract SponsorPaymaster is Initializable, BasePaymaster, UUPSUpgradeable, ReentrancyGuard, ISponsorPaymaster {
     using SafeERC20 for IERC20;
 
-    // ========== Events ============
-    event AppRegistrySet(address appRegistry, address _oldRegistry);
-
     // calculated cost of the postOp
     uint256 public constant COST_OF_POST = 200_000;
     uint256 public constant MAX_COST_OF_VERIFICATION = 230_000;
     uint256 public constant MAX_COST_OF_PREVERIFICATION = 110_000;
-    uint256 public constant MAX_COST_OF_USEROP = 0.03 ether;
 
     uint256 public constant RATE_LIMIT_PERIOD = 1 minutes;
     uint256 public constant RATE_LIMIT_THRESHOLD_TOTAL = 50;
 
+    uint256 public userOpMaxCost;
```

Comment:
I would set the initial value here because sponsor paymaster is already initialized so the initializer is not going to be called

Reply:
we can't do that when using UUPS...
https://docs.openzeppelin.com/upgrades-plugins/1.x/writing-upgradeable#avoid-initial-values-in-field-declarations

Reply:
also, [this](https://docs.openzeppelin.com/upgrades-plugins/1.x/writing-upgradeable#modifying-your-contracts) is interesting. when upgrading a contract:

- you **cannot** change the order in which the contract state variables are declared
- you **cannot** change the type of a variable
- you **cannot** change the type of a variable
- you **cannot** introduce a new variable before existing ones
- you **cannot** remove an existing variable

If you need to introduce a new variable, make sure you always do so at the end:

we may need to re-visit our last changes on the contracts and maybe revert the changes (if there are) and re-deploy? 
we should ideally also make sure that all tests run locally but also on mainnet fork (on the deployed contracts)

Reply:
Yes, I have been taking care to follow this. It is what we did at Babylon. Always add at the end

you cannot change the order in which the contract state variables are declared
you cannot change the type of a variable
you cannot introduce a new variable before existing ones
you cannot remove an existing variable